### PR TITLE
feat: add `TypeConstructor.SymmetricDiff`

### PR DESCRIPTION
**Pull request by magnus-madsen**

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/fmt/SimpleType.scala [hunk 1]
```diff
@@ -551,6 +556,11 @@ object SimpleType {
             case _ :: _ :: _ :: _ => throw new OverAppliedType(t.loc)
           }
 
+        case TypeConstructor.SymmetricDiff =>
+          // collapse into a chain of symmetric differences
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/fmt/SimpleType.scala:560]: Not really collapsing but ok

File: main/src/ca/uwaterloo/flix/language/ast/Purity.scala [hunk 2]
```diff
@@ -159,6 +159,11 @@ object Purity {
       val t1 = evaluateFormula(tpe1)
       val t2 = evaluateFormula(tpe2)
       t1.intersect(t2)
+    case Type.Apply(Type.Apply(Type.Cst(TypeConstructor.SymmetricDiff, _), tpe1, _), tpe2, _) =>
+      val t1 = evaluateFormula(tpe1)
+      val t2 = evaluateFormula(tpe2)
+      // a ⊕ b = (a ∪ b) - (a ∩ b) = (a ∪ b) ∩ ¬(a ∩ b) = (a ∪ b) ∩ (U - (a ∩ b))
+      t1.union(t2).intersect(universe.diff(t1.intersect(t2)))
```
Comment (mlutze) [inline main/src/ca/uwaterloo/flix/language/ast/Purity.scala:166]: Why not `t1.union(t2).diff(t1.intersect(t2))`?

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/ast/Purity.scala:166]: No diff on types.

Comment (mlutze) [reply main/src/ca/uwaterloo/flix/language/ast/Purity.scala:166]: `t1.union(t2)` is a set, no?

File: main/src/ca/uwaterloo/flix/language/fmt/SimpleType.scala (hunk 1 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/fmt/SimpleType.scala:560]: Fixed

File: main/src/ca/uwaterloo/flix/language/ast/Purity.scala (hunk 2 above)
Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/ast/Purity.scala]: Fixed.

**Pull request merged**